File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/RetrofitRoom/Repository/PopulateDbAsync.java
package com.andr.mvvm.RetrofitRoom.Repository;

import android.os.AsyncTask;

import com.andr.mvvm.repository.User;
import com.andr.mvvm.repository.UserDao;
import com.andr.mvvm.repository.UserRoomDataBase;

public class PopulateDbAsync extends AsyncTask<Void, Void, Void> {

    private final PostInfoDao mDao;

    PopulateDbAsync(PostInfoRoomDataBase db) {
        mDao = db.postInfoDao();
    }

    @Override
    protected Void doInBackground(final Void... params) {
        mDao.deleteAll();
       /* User user = new User("Chandra","SW");
        mDao.insert(user);
        user = new User("Mohan","student");
        mDao.insert(user);*/
        return null;
    }
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Kotlin_Retrofit_Room/Repository/PopulateDBAsync.kt
package com.mvvm.kot.Kotlin_Retrofit_Room.Repository

import android.os.AsyncTask
import com.mvvm.kot.ViewModels.User

    class PopulateDbAsync internal constructor(db: PostInfoRoomDataBase) : AsyncTask<Void, Void, Void>() {

        private val mDao: PostInfoDao

        init {
            mDao = db.postInfoDao()
        }

        override fun doInBackground(vararg params: Void): Void? {
            mDao.deleteAll()
            return null
        }
    }
